mathics/builtin/makeboxes.py

```diff
@@ -502,15 +526,36 @@ def get_op(op):
 
         elements = expr.elements
         if len(elements) > 1:
-            if h.has_form("List", len(elements) - 1):
-                ops = [get_op(op) for op in h.elements]
+            if operator.has_form("List", len(elements) - 1):
+                operator = [format_operator(op) for op in operator.elements]
+                return make_boxes_infix(elements, operator, precedence, grouping, form)
             else:
-                ops = [get_op(h)] * (len(elements) - 1)
-            return make_boxes_infix(elements, ops, precedence, grouping, f)
+                encoding_rule = evaluation.definitions.get_ownvalue(
+                    "$CharacterEncoding"
+                )
+                encoding = (
```

Comment:
Again, according to the experiments I have run, `MakeBoxes` rules should be agnostic about the encoding. The encoding should be take into account when the Boxes are "rendered" by a consumer (the command line, the function that implements ToString, or the exporter.)

Reply:
Right now, we don't have a formatting phas for this for infix operators and probably not for any operators in general.

What was done here was to push this back one small level from making formatting decisions on infix operators inside static MakeBox rules to the evaluation of the MakeBox. 

Furthermore, moving this to a separate formatting phase looks like it will be major work. 

What I have done here though was to _mark_ with FIXME the places that violate the Eval, Box, Format process for later.
So in that sense this is a step and direction towards where this needs to go.

As I said, given the size of this PR, and the amount of work that still needs to be done, not just here but across the board. my take is to go with this for now, and give up on further Form work, until we have better isolated Boxing routines, and can start isolating more formatting routines on something simpler.

Reply:
Also, I would like to note that the other big thing of these PRs is that we have started to separate properties of characters and operators and pull these in from mathics scanner.   In particular using character tables we can start to represent translations of operators to Unicode, and ASCII. (WL-Unicode, and AMSLaTeX could easily follow).

There is  more work to do here too. For example operator precedence should come from mathics scanner. However this is a start.

In sum, the amount of refactoring work is quote huge and I do not think can be done in one step. However, there is a vision or direction outlined here, a step made towards that, and FIXME's for where there is more to do.

Reply:
OK, then go ahead. Merge it when you feel ready.